src/render_keyhook.c

```diff
@@ -0,0 +1,28 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   render_keyhook.c                                   :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: gwolf <gwolf@student.42vienna.com>         +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/12/19 22:40:44 by gwolf             #+#    #+#             */
+/*   Updated: 2023/12/19 22:49:38 by gwolf            ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "render.h"
+
+int	ft_keyhook_press(int key, t_render *render)
+{
+	static bool	print;
+
+	if (key == K_ESC)
+		mlx_loop_end(render->mlx_ptrs.mlx_ptr);
+	else if (key == K_C && !print)
+	{
+		print = true;
```

Comment:
so it doesn't get outputted repeatedly I presume

Reply:
Yes, if pressed again while the function prints, the file gets corrupted